### ISSUE TITLE
NVDA fails to properly gain focus when opening Terminal

### ISSUE THREAD

**Issue by Neurrone** [nvaccess/nvda#13130]
/CC @codeofdusk 

### Steps to reproduce:

Open Windows Terminal

### Actual behavior:

Sometimes, NVDA says "Windows Powershell" and doesn't say "terminal" like it usually does. Incoming text is not read out at all. Restarting NVDA does not fix this issue, only restarting Terminal does.

Here's dev info when pressing NVDA+f1 when this happens:

```
name: 'Windows PowerShell'
role: Role.PANE
roleText: None
states: State.FOCUSABLE
isFocusable: True
hasFocus: False
Python object: <NVDAObjects.IAccessible.IAccessible object at 0x066FFB90>
Python class mro: (<class 'NVDAObjects.IAccessible.IAccessible'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'garbageHandler.TrackedObject'>, <class 'object'>)
description: None
location: RectLTWH(left=190, top=190, width=1730, height=949)
value: None
appModule: <'appModuleHandler' (appName 'windowsterminal', process ID 14996) at address 7dc8f0>
appModule.productName: 'Microsoft.WindowsTerminal'
appModule.productVersion: '1.11.2921.0'
TextInfo: <class 'NVDAObjects.NVDAObjectTextInfo'>
windowHandle: 854426
windowClassName: 'CASCADIA_HOSTING_WINDOW_CLASS'
windowControlID: 0
windowStyle: 349110272
extendedWindowStyle: 256
windowThreadID: 19768
windowText: 'Windows PowerShell'
displayText: ''
IAccessibleObject: <POINTER(IAccessible) ptr=0xb143ee0 at 2e21a30>
IAccessibleChildID: 0
IAccessible event parameters: windowHandle=854426, objectID=-4, childID=0
IAccessible accName: 'Windows PowerShell'
IAccessible accRole: ROLE_SYSTEM_CLIENT
IAccessible accState: STATE_SYSTEM_FOCUSABLE, STATE_SYSTEM_VALID (1048576)
IAccessible accDescription: None
IAccessible accValue: None
```
### Expected behavior:

The ability to use Terminal as usual

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
alpha-24184,fd9ae8c4

#### Windows version:
Windows 10.0.19043.1348

#### Name and version of other software in use when reproducing the issue:
Windows Terminal 1.11.2921.0

#### Does the issue still occur after restarting your computer?
Yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
Yes, this behaviour shows up for me even across multiple NVDA versions

#### If NVDA add-ons are disabled, is your problem still occurring?
Unknown

#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
N/A

**Comment by codeofdusk** [comment]
Cc @carlos-zamora.

**Comment by carlos-zamora** [comment]
Is there a way to see what object in the UI tree actually has focus here? For reference, the UI tree for Windows Terminal should look something like this:
- Window "profile name"
   - pane "DesktopWindowXamlSource"
      - pane "empty name"
         - terminal "profile name"

I suspect that the focus might have gotten stuck on one of the middle two panes, in which case, we may have to loop the XAML team into this.

@zadjii-msft have you seen anything like this or relevant over in the https://github.com/microsoft/terminal/ repo?

**Comment by codeofdusk** [comment]
Also see `Python object: <NVDAObjects.IAccessible.IAccessible object at 0x066FFB90>`. This means that NVDA is falling back to MSAA here, probably because `HasServerSideProvider` for the window in question is returning `False`.

**Comment by zadjii-msft** [comment]
Might be https://github.com/microsoft/terminal/issues/10911 (which OP also filed originally(!))

**Comment by Neurrone** [comment]
Not sure if they are the same. For the linked issue, focus does appear to be correctly set to the terminal. I'll confirm this the next time it isn't reading incoming text.

**Comment by Neurrone** [comment]
This is probably not related to microsoft/terminal#10911. Based on what gets spoken when switching to Terminal, focus seems to be set correctly, except that incoming text is just now read at all.

Here's the developer info.

```
Developer info for navigator object:
name: 'Windows PowerShell'
role: Role.TERMINAL
roleText: None
states: State.FOCUSABLE, State.FOCUSED
isFocusable: True
hasFocus: True
Python object: <NVDAObjects.Dynamic_WinTerminalUIAEditableTextWithAutoSelectDetectionUIA object at 0x012D7D30>
Python class mro: (<class 'NVDAObjects.Dynamic_WinTerminalUIAEditableTextWithAutoSelectDetectionUIA'>, <class 'NVDAObjects.UIA.winConsoleUIA.WinTerminalUIA'>, <class 'NVDAObjects.behaviors.EnhancedTermTypedCharSupport'>, <class 'NVDAObjects.behaviors.Terminal'>, <class 'NVDAObjects.behaviors.LiveText'>, <class 'NVDAObjects.behaviors.EditableTextWithAutoSelectDetection'>, <class 'NVDAObjects.behaviors.EditableText'>, <class 'editableText.EditableText'>, <class 'NVDAObjects.UIA.UIA'>, <class 'NVDAObjects.window.Window'>, <class 'NVDAObjects.NVDAObject'>, <class 'documentBase.TextContainerObject'>, <class 'baseObject.ScriptableObject'>, <class 'baseObject.AutoPropertyObject'>, <class 'garbageHandler.TrackedObject'>, <class 'object'>)
description: '~/source'
location: RectLTWH(left=0, top=47, width=1920, height=973)
value: None
appModule: <'appModuleHandler' (appName 'windowsterminal', process ID 16416) at address 12cbf30>
appModule.productName: 'Microsoft.WindowsTerminal'
appModule.productVersion: '1.11.2921.0'
TextInfo: <class 'NVDAObjects.UIA.UIATextInfo'>
windowHandle: 1442620
windowClassName: 'Windows.UI.Input.InputSite.WindowClass'
windowControlID: 0
windowStyle: 1342177280
extendedWindowStyle: 0
windowThreadID: 16664
windowText: ''
displayText: ''
UIAElement: <POINTER(IUIAutomationElement) ptr=0x8b2f5c8 at 12ba170>
UIA automationID: 
UIA frameworkID: XAML
UIA runtimeID: (42, 1442620, 4, 2)
UIA providerDescription: [pid:16416,providerId:0x0 Main(parent link):Unidentified Provider (unmanaged:Windows.UI.Xaml.dll)]
UIA className: TermControl
UIA patterns available: ScrollItemPattern, TextPattern, LegacyIAccessiblePattern
diffing algorithm: <diffHandler.DiffMatchPatch object at 0x04B408D0>
```

**Comment by Adriani90** [comment]
@Neurrone are you still having this issue in NVDA 2024.4.1?

**Comment by Neurrone** [comment]
This doesn't seem to happen for me anymore.
